mypy/meet.py

```diff
@@ -90,10 +90,15 @@ def meet_types(s: Type, t: Type) -> ProperType:
             return t
 
     if not isinstance(s, UnboundType) and not isinstance(t, UnboundType):
+        swap = isinstance(t, TupleType) and not isinstance(s, TupleType)
+        if swap:
+            s, t = t, s
```

Comment:
This looks suspicious, the check below is for _proper_ subtype, that should not now anything about `Any`.